test/e2e/azure_apiserver_ilb.go

```diff
@@ -0,0 +1,463 @@
+                                             |
+                                             v
+                                  +--------------------------------+
+                                  |        Test Complete           |
+                                  +--------------------------------+
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/utils/ptr"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+)
+
+// AzureAPIServerILBSpecInput is the input for AzureAPIServerILBSpec.
+type AzureAPIServerILBSpecInput struct {
+	BootstrapClusterProxy                   framework.ClusterProxy
+	Cluster                                 *clusterv1.Cluster
+	Namespace                               *corev1.Namespace
+	ClusterName                             string
+	ExpectedWorkerNodes                     int32
+	WaitIntervals                           []interface{}
+	TemplateHasPrivateIPCustomDNSResolution bool
+}
+
+// AzureAPIServerILBSpec implements a test that verifies the Azure API server ILB is created.
+func AzureAPIServerILBSpec(ctx context.Context, inputGetter func() AzureAPIServerILBSpecInput) {
+	var (
+		specName = "azure-apiserver-ilb"
+		input    AzureAPIServerILBSpecInput
+	)
+
+	input = inputGetter()
+	Expect(input.Namespace).NotTo(BeNil(), "Invalid argument. input.Namespace can't be nil when calling %s spec", specName)
+	Expect(input.ClusterName).NotTo(BeEmpty(), "Invalid argument. input.ClusterName can't be empty when calling %s spec", specName)
+
+	By("Fetching new Azure Credentials")
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Getting azureLoadBalancerClient")
+	azureLoadBalancerClient, err := armnetwork.NewLoadBalancersClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Verifying the Azure API Server Internal Load Balancer is created")
+	groupName := os.Getenv(AzureResourceGroup)
+	fmt.Fprintf(GinkgoWriter, "Azure Resource Group: %s\n", groupName)
+	internalLoadbalancerName := fmt.Sprintf("%s-%s", input.ClusterName, "public-lb-internal")
+
+	backoff := wait.Backoff{
+		Duration: 300 * time.Second,
+		Factor:   0.5,
+		Jitter:   0.5,
+		Steps:    7,
+	}
```

Comment:
nit: Might be good to set these to these established variables: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/azure_privatecluster.go#L179

Reply:
If you need a different value for them, it could be good to at least set it as a new variable

Reply:
Thanks for the linking https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/azure_privatecluster.go#L179. 

I purposefully created a new local variable, `backoff`,  with my own _**readable values**_ because the hardcoded values, mentioned below, https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/61c6f3f180539609090fbf5a943b2e021cfbf129/test/e2e/azure_privatecluster.go#L178-L183 
are not readable. 

For instance, unless I click and navigate to `retryBackoffInitialDuration` declaration, I do not know that it is set to `100 * time.Millisecond`. 

The better approach would be to rename `retryBackoffInitialDuration` -> `retryBackoff_100MilSeconds` or something that makes reading the code easier.

Reply:
> If you need a different value for them, it could be good to at least set it as a new variable

Can you please elaborate ?
Would you like me to create a new local variable for the Duration values ? 
Or rename the variable `backoff` ? 
Since `backoff` is a locally scoped variable that is usually changing per retry func, do we really need to standardize the values for Duration, Jitter... etc ?

Reply:
Okay that's fair! There isn't a need to standardize them, and if it is more readable the way you have it, I think it's best to leave it as-is.